src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -318,22 +319,104 @@ private static boolean isAnonInnerClassInit(DetailAST lcurly) {
     }
 
     /**
-     * Collects validation details.
-     * @param ast detail ast.
-     * @return object that contain all details to make a validation.
+     * Checks if definition body is empty.
+     * @param lcurly left curly.
+     * @return true if definition body is empty.
+     */
+    private static boolean isEmptyBody(DetailAST lcurly) {
+        boolean result = false;
+        if (lcurly.getParent().getType() == TokenTypes.OBJBLOCK) {
+            if (lcurly.getNextSibling().getType() == TokenTypes.RCURLY) {
+                result = true;
+            }
+        }
+        else if (lcurly.getFirstChild().getType() == TokenTypes.RCURLY) {
+            result = true;
+        }
+        return result;
+    }
+
+    /**
+     * Finds next token after the given one.
+     * @param ast the given node.
+     * @return the token which represents next lexical item.
+     */
+    private static DetailAST getNextToken(DetailAST ast) {
+        DetailAST next = null;
+        DetailAST parent = ast;
+        while (next == null) {
+            next = parent.getNextSibling();
+            parent = parent.getParent();
+        }
+        return CheckUtils.getFirstNode(next);
+    }
+
+    /**
+     * Checks if right curly has line break before.
+     * @param rightCurly right curly token.
+     * @return true, if right curly has line break before.
+     */
+    private static boolean hasLineBreakBefore(DetailAST rightCurly) {
+        final DetailAST previousToken = rightCurly.getPreviousSibling();
+        return previousToken == null
+                || rightCurly.getLineNo() != previousToken.getLineNo();
+    }
+
+    /**
+     * Structure that contains all details for validation.
      */
-    // -@cs[JavaNCSS] getDetails() method is a huge SWITCH, it has to be monolithic
-    // -@cs[ExecutableStatementCount] getDetails() method is a huge SWITCH, it has to be monolithic
-    // -@cs[NPathComplexity] getDetails() method is a huge SWITCH, it has to be monolithic
-    private static Details getDetails(DetailAST ast) {
```

Comment:
This function was exactly at the thresholds for the complexity checks and statement counts. So, had to restructure it. Also restructuring might prove to be useful in the future when more tokens might need to be added to the check